k2/python/k2/fsa_algo.py

```diff
@@ -50,37 +51,55 @@ def top_sort(fsa: Fsa) -> Fsa:
     for name, value in fsa.named_tensor_attr():
         setattr(sorted_fsa, name, value.index_select(0, arc_map))
     for name, value in fsa.named_non_tensor_attr():
-        setattr(sorted_fsa, name, deepcopy(value))
+        setattr(sorted_fsa, name, value)
     return sorted_fsa
 
 
 def intersect(a_fsa: Fsa, b_fsa: Fsa) -> Fsa:
     '''Compute the intersection of two FSAs.
 
-    Note:
-      Intersection is also known as compose in FST.
-
     Args:
       a_fsa:
         The first input FSA. It can be either a single FSA or a vector of FSAs.
       b_fsa:
         The second input FSA. It can be either a single FSA or a vector of FSAs.
 
+    CAUTION:
+      ``aux_labels`` are discarded since it is not clear how to set the
+      aux_labels of the output FSA.
+
     Returns:
       The result of intersecting a_fsa and b_fsa.
     '''
+    if not a_fsa.arc_labels_sorted:
```

Comment:
I'd rather have the user explicitly do the arc sorting, like it is done in OpenFST.
When we test whether an Fsa is arc-sorted, it will be done by:
   - if the properties are cached, use the cached properties;  otherwise compute and cache the properties first.

Reply:
The problem is that we are calling `invert_()` inside `k2.compose()` and `arc_sort()`
is called after `invert_()`. Users do not have the chance to call `arc_sort` before `k2.compose()`.

Reply:
Sounds it would be simpler by removing `compose`... we just support `intersect`, uses should call `invert` if they need.

Reply:
`compose` is removed. We give more freedom to users.